packages/ember-application/tests/system/visit_test.js

```diff
@@ -30,59 +73,347 @@ if (isEnabled('ember-application-visit')) {
   // instance initializer that would normally get called on DOM ready
   // does not fire.
   QUnit.test('Applications with autoboot set to false do not autoboot', function(assert) {
-    QUnit.expect(1);
+    QUnit.expect(4);
     QUnit.stop();
 
+    let appBooted = 0;
+    let instanceBooted = 0;
+
     run(function() {
-      var app = createApplication();
+      createApplication();
 
-      // Start the timeout
-      var timeout = setTimeout(function() {
-        ok(true, '500ms elapsed without initializers being called');
-        QUnit.start();
-      }, 500);
+      App.initializer({
+        name: 'assert-no-autoboot',
+        initialize() {
+          appBooted++;
+        }
+      });
+
+      App.instanceInitializer({
+        name: 'assert-no-autoboot',
+        initialize() {
+          instanceBooted++;
+        }
+      });
+    });
+
+    // Continue after 500ms
+    setTimeout(function() {
+      QUnit.start();
+      ok(appBooted === 0, '500ms elapsed without app being booted');
+      ok(instanceBooted === 0, '500ms elapsed without instances being booted');
+
+      QUnit.stop();
+
+      run(function() {
+        App.boot().then(
+          () => {
+            QUnit.start();
+            assert.ok(appBooted === 1, 'app should boot when manually calling `app.boot()`');
+            assert.ok(instanceBooted === 0, 'no instances should be booted automatically when manually calling `app.boot()');
+          },
+          (error) => {
+            QUnit.start();
+            assert.ok(false, 'the boot process failed with ' + error);
+          }
+        );
+      });
+    }, 500);
```

Comment:
I inherited this from the existing code, so I don't feel comfortable making the call to change it; but you are welcomed to tell me what to do ;)